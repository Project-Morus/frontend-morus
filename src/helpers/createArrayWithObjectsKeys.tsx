export const createArrayWithObjectsKeys = (quantity: number) => {
  const dotsKeys = []

  for (let i = 1; i <= quantity; i++) {
    [...dotsKeys, dotsKeys.push({ key: i })]
  }

  return dotsKeys
}