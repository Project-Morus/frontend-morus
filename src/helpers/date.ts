export const date = new Date()

export function formatDate(date: string) {
  const convertDate = new Date(date)

  return new Intl.DateTimeFormat('pt-BR').format(convertDate)
}

export function currentDateFormatted() {
  return new Intl.DateTimeFormat('pt-BR', { timeZone: 'UTC' }).format(date)
}