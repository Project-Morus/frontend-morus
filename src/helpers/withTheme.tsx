import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

