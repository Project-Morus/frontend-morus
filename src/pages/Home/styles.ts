import styled from "styled-components";

export const ContainerSC = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.spacing[4]};
`