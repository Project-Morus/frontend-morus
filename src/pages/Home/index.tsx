import { useEffect, useState } from "react";
import CardInformations from "./parts/CardInformations";

import { CardInformationProps } from "./parts/types";
import httpClient from "../../services/httpClient";
import { ContainerSC } from "./styles";


const Home = () => {
  const [data, setData] = useState<CardInformationProps[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  useEffect(() => {
    async function fetchData() {
      try {
        console.log("entrou no try");
        const response = await httpClient.get("api/ListarInformacao");
        setData(response.data.data);
        setIsLoading(false);
        console.log(response.data.data);
      } catch (error) {
        console.log("entrou no catch");
        console.error("Erro ao buscar dados da API:", error);
        setIsLoading(false);
      }
    }

    fetchData();
  }, []);

  const renderList = () =>
    data.map((item: CardInformationProps) => {
      return <CardInformations titulo={item.titulo} descricao={item.descricao} />;
    });

  return <ContainerSC>{renderList()}</ContainerSC>;
};

export default Home;
