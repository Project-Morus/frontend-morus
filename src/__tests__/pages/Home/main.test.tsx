import { render, screen } from '@testing-library/react';
import Home from '../../../pages/Home';
import { withTheme } from '../../../helpers/withTheme';

const HomeWithTheme = withTheme(Home)
describe('Home Component', () => {
  test('Render component', () => {
    render(<HomeWithTheme />)
    expect(screen.getByText('Home'))
  })

  test('Snapshot Home', () => {
    const { asFragment } = render(
      <HomeWithTheme />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})