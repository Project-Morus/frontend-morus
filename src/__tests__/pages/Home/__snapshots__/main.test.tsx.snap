// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot Home 1`] = `
<DocumentFragment>
  <div>
    <h1
      style="font-size: 2.1875rem;"
    >
      Home
    </h1>
  </div>
</DocumentFragment>
`;
